public/app/features/explore/utils/links.ts

```diff
@@ -127,10 +128,34 @@ export const getFieldLinksForExplore = (options: {
         }
         return linkModel;
       } else {
+        let internalLinkSpecificVars: ScopedVars = {};
+        if (link.internal?.transformations) {
+          const fieldValue = field.values.get(rowIndex);
+          link.internal?.transformations.forEach((transformation) => {
+            if (transformation.type === 'regex' && transformation.expression) {
```

Comment:
Should we implement it in a cleaner way? This has the potential to grow quite a bit, maybe break the logic for transformations out to its own file?

Reply:
We will defo need it in the future and maybe move it to `feature/correlations` but we can do it later. Also you can do it now as it may make it easier to test just the transformation logic.